reactor-stream/src/main/java/reactor/rx/action/transformation/GroupByAction.java

```diff
@@ -160,9 +160,7 @@ protected void doComplete() {
 			stream.onComplete();
 		}
 
-		if (groupByMap.isEmpty()) {
-			broadcastComplete();
-		}
+		super.doComplete();
```

Comment:
That will only broadcast complete if no current groupBy sub-stream is currently listening as the partition can themselves trigger this complete too.